dmscripts/generate_framework_master_csv.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+"""Class used to output a framework master csv."""
+import csv
+from sys import stdout
+
+
+class GenerateMasterCSV(object):
+    """"Generate a master csv given a framework."""
+
+    static_fieldnames = ['supplier_id', 'supplier_dm_name', 'application_status', 'declaration_status']
+    lot_name_prefixes = ['draft', 'completed']
```

Comment:
I'd specify these in a slightly different way - what you're actually listing here is human-readable equivalents of certain service statuses. So to do it in a slightly more declarative way I'd do:

```
service_statuses_labels = {
    "submitted": "completed",
    "not-submitted": "draft",
}
```

and then when you use this in `_update_with_lot_data`, loop through it and do it in a data-driven way. Actually you might need to use an `OrderedDict` to make the output deterministic. Or maybe not use a dict at all, just a tuple of `(status, label,)` pairs, seeing as you don't actually have to do a lookup by key by the looks of it.